### DISCUSSION TITLE
PSAT CLI tool with message after update to 0.8.0 version

### DISCUSSION THREAD

**Discussion by Joeyindigital** [GoogleChromeLabs/ps-analysis-tool#733] [Support Forum] [answered]
just update the new version of cli psat tool. While I run npm cli run -- -u https://bbc.com -ab, this message appears. 
"Please provide one and only one of the following
        a) URL of a site (-u or --url)
        b) URL of a sitemap (-s or --sitemap-url)
        c) Path to a CSV file (-c or --csv-path)
        d) Path to an XML file (-p or --sitemap-path)"
Any reason why this is happening after the updates?
        
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/20696a2b-4c87-43bc-a7e8-8b033dd55ba2)

**Comment by milindmore22** [comment]
Hello @Joeyindigital 

Thank you for reaching out to us, we checked your screenshots and found that the output you provided shows 
`> node dist/cli/index.js https://bbc.com/`, but it should ideally be `> node dist/cli/index.js -u https://bbc.com/ -ab`.

Unfortunately, we couldn’t replicate the issue on our end, so we’re not entirely sure what caused it.

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/1132541/0ab605c2-f9c7-409d-b3c1-23909a050560)

**To diagnose the issue further, please share some additional information?**
1. **Operating System (OS) and Version:** This helps us understand if there might be any OS-specific quirks affecting the CLI options.
2. **PowerShell Version :**  You can find instructions on how to check the PowerShell version using this helpful guide: [How to Check the PowerShell Version in Windows 10](https://www.howtogeek.com/731885/how-to-check-the-powershell-version-in-windows-10/)
3. **Node Version:** You can check your Node.js version by running the command `node -v` in your terminal.

**Meanwhile, here are some general suggestions that might help:**

- Reinstall and rebuild the project by running:
  - `npm i`
  - `npm run cli:build`

- After the rebuild, run the command for analysis again using:

  - `npm run cli -- -u https://bbc.com/ -ab`

- If the CLI options are still missing in the output, try using the alternative CLI options:
  - `npm run cli -- --url https://bbc.com/ --accept-banner`

We hope this helps! Please let us know if you have any other questions.

**Comment by Joeyindigital** [comment]
I have reinstalled the package and still shows the same issue. 
(I used node v22.1.0 with psat cli 0.8.0 with windows 11)
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/31fd5f56-983e-4dbe-9681-aac70842f6f6)

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/3d299ae0-19ad-43e2-b08c-6a9e8cf114ce)

**Comment by Joeyindigital** [comment]
And I tested it on command prompt, I received this error.
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/166688925/3372d385-a0c3-461a-a52d-85e29862ef73)
"node:internal/child_process:421
    throw new ErrnoException(err, 'spawn');
          ^

Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at spawn (node:child_process:760:9)
    at C:\Users\JoeyHsin-YinHou(OMG)\ps-analysis-tool\dist\cli\index.js:71:31
    at Generator.next (<anonymous>)
    at C:\Users\JoeyHsin-YinHou(OMG)\ps-analysis-tool\dist\cli\index.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\JoeyHsin-YinHou(OMG)\ps-analysis-tool\dist\cli\index.js:4:12)
    at startDashboardServer (C:\Users\JoeyHsin-YinHou(OMG)\ps-analysis-tool\dist\cli\index.js:70:45)
    at C:\Users\JoeyHsin-YinHou(OMG)\ps-analysis-tool\dist\cli\index.js:172:5
    at Generator.next (<anonymous>) {
  errno: -4071,
  code: 'EINVAL',
  syscall: 'spawn'
}"

**Comment by milindmore22** [comment]
Hello @Joeyindigital

Thank you for sharing the details, we tried Node.js **v22.1.0** and encountered a similar bug.

![image (1)](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/1132541/c8d86f1b-3dff-43b5-8b1f-1fdb381087c6)

While we’re working on a fix, we recommend switching to Node.js version **v18.1** for now.

![Screenshot_win11_2024-05-16_15_05_10](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/1132541/d1ec1628-db34-471c-a022-915ca116bd1b)


You can easily install and switch between different Node.js versions using nvm. This [article](https://www.freecodecamp.org/news/nvm-for-windows-how-to-download-and-install-node-version-manager-in-windows-10/) provides a helpful guide.

**Comment by milindmore22** [comment]
Hello @Joeyindigital 

I'm checking in to see if you've been able to use the **PSAT CLI** after switching to **Node.js version 18.1.**

Please reach out to me if you experience any issues.

**Comment by milindmore22** [comment, marked as answer]
Hi @Joeyindigital,

I'm excited to announce the release of PSAT version [**0.9.0**](https://github.com/GoogleChromeLabs/ps-analysis-tool/discussions/732)! This new version is compatible with Node.js version 18 or higher, solving the compatibility issue you experienced with the node version v22.1.0.

To get started:

1. Pull the latest changes using `git pull`.
2. Rebuild the CLI with `npm ci` followed by `npm run build:all`.
3. Once the build is complete, `run npm run cli -- -u https://bbc.com`. This will generate an HTML file report that you can open in your browser.

To learn more about PSAT CLI, please visit our updated [wiki page](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Command-Line-Interface). 

Please let me know if you need any further assistance.

**Comment by milindmore22** [comment]
I will be converting this issue to support forum discussion, For future reference, please [create support discussion](https://github.com/GoogleChromeLabs/ps-analysis-tool/discussions/new?category=support-forum) if you have any question about the PSAT.
